jupyterhub/auth.py

```diff
@@ -87,7 +94,28 @@ class LocalAuthenticator(Authenticator):
         should I try to create the system user?
         """
     )
-    
+
+    whitelist_group = Unicode(
+        config=True,
+        help="Automatically whitelist anyone in this group.",
+    )
+
+    def check_whitelist(self, username):
+        return (
+            super().check_whitelist(username) or
```

Comment:
In the case where the user whitelist is empty, `check_whitelist_group` will never be called.

For simplicity, I might require whitelisting by name or group to be mutually exclusive. What do you think?